scripts/core/editor3/actions/toolbar.tsx

```diff
@@ -66,13 +67,36 @@ export function removeFormat() {
  */
 export function insertMedia(files?, targetBlockKey = null) {
     const superdesk = ng.get('superdesk');
+    const renditions = ng.get('renditions');
+
+    const editedMedia = [];
 
     return (dispatch) => {
-        superdesk.intent('upload', 'media', files).then((media) => {
-            dispatch({
-                type: 'TOOLBAR_INSERT_MEDIA',
-                payload: {files: media, targetBlockKey: targetBlockKey},
-            });
+        superdesk.intent('upload', 'media', files).then((media_list) => {
```

Comment:
It's better to not modify function arguments / promise results, especially when it's easy to avoid like here. The function that returns promises might be using the same references for caching or other purposes and you may break something by modifying it directly. Create a `const media_list_copy = [...media_list];` and remove items from that instead.